dogm/demo/utils/include/metrics.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2020 Michael Koesel and respective contributors
+// SPDX-License-Identifier: MIT
+// See accompanying LICENSE file for detailed information
+
+#ifndef METRICS_H
+#define METRICS_H
+
+#include "precision_evaluator.h"
+
+class Metric
+{
+public:
+    Metric() : cumulative_error{}, number_of_detections(0) {}
+    virtual ~Metric(){};
+
+    virtual void reset()
+    {
+        cumulative_error = {};
+        number_of_detections = 0;
+    }
+    virtual PointWithVelocity update(const PointWithVelocity& cluster_mean, const Vehicle& vehicle) { return {}; }
```

Comment:
These functions should never be used, correct? Consider (also for `compute`)
```suggestion
    virtual PointWithVelocity update(const PointWithVelocity& cluster_mean, const Vehicle& vehicle) = 0;
```

This will make it a pure abstract function and prohibit direct instances of `Metric`.